UPGRADE.md

```diff
@@ -0,0 +1,131 @@
+# Phoenix 1.5.x to 1.6 upgrade instructions
+
+## Update your deps
+
+In `mix.exs`, update your `phoenix`, `phoenix_html`, `telemetry_metrics`, `telemetry_poller` and `phoenix_live_dashboard` deps, and add `phoenix_live_view`:
+
+```elixir
+def deps do
+    [
+      {:phoenix, "~> 1.6.0"},
+      ...
+      {:phoenix_html, "~> 3.0"},
+      {:phoenix_live_view, "~> 0.16.4"},
+      {:phoenix_live_dashboard, "~> 0.5"},
+      {:telemetry_metrics, "~> 0.6"},
+      {:telemetry_poller, "~> 0.5"},
+      ...
+    ]
+end
+```
+
+Next, run `mix deps.get` to grab your new deps.
+
+
+## Rename your `.html.eex` and `.html.leex` templates to `.html.heex` (optional)
+
+While leex templates have been deprecated, this step is optional. For the most part, existing templates should continue to work, but the HTML-aware HEEx engine will enforce valid markup and is more strict in the elixir expressions that appear within an open and closing tag.
+For example, the following code will raise:
+
+```eex
+<div id="<%= @id %>">
+```
+
+Instead of the standard `<%= %>` EEx expressions, elixir expressions inside tags can only appear withing `{}`, such as:
+
+```eex
+<div id={@id}>
+```
+
+`<%= %>` expressions remain valid outside of HTML tags in the EEx engine.
+
+To update your existing templates, rename all your `.html.eex` and `.html.leex` templats to `.html.heex` and follow the parser errors to find any tags that require the new `{}` attribute form.
+
+Also be sure to review the [HEEx documentation](TODO) for more information on features.
+
+## Migrate to esbuild for js and css bundling (optional)
+
+Phoenix's `watchers` configuration is build-tool agnostic, so you may continue to enjoy your existing webpack configurations generated by phoenix 1.5 or earlier. If only have basic js and css needs and you would like to take advantage of our new `esbuild` usage, for a dependency-free asset builder powered by a portably binary, follow these steps:
+
+First delete your webpack config and related node files:
+
+```console
+$ rm assets/webpack.config.js assets/package.json assets/package-lock.json assets/.babelrc
+$ rm -rf assetes/node_modules
```

Comment:
```suggestion
$ rm -rf assets/node_modules
```

Originally reported at https://gist.github.com/chrismccord/2ab350f154235ad4a4d0f4de6decba7b#gistcomment-3873576

Reply:
Yeah, I noticed that and opened a different PR #4586 for this.

Reply:
Ah, okay :+1: I didn't notice that one. I guess it makes sense to open them separately.

Reply:
Doesn't it mean that PR #4586 supersedes this one, and there's no need to merge this one?